docs/website/contents/for-developers/CivicTime.md

```diff
@@ -0,0 +1,193 @@
+
+An honest node will not fetch a block before validating the corresponding header, so the above rule prevents a node from receiving a block before the wall clock has reached the onset of its slot.
+
+*Aside*.
+It is possibly that a block from the future is already in the database when the node starts.
+That's a corner case that seems unlikely to matter in practice.
+
+## Time Translations
+
+The Ouroboros Praos protocol and the Cardano details built around it are almost exclusively defined in terms of slots rather than common civic measures of time, such as POSIX time, UTC, etc.
+Each block explicitly inhabits some slot.
+The same is true for block headers and the election proofs therein.
+For the sake of determinism, transactions are labeled with a range of slots in which the transaction can be valid.
+
+The are only two exceptions.
+
+- The node's (commodity) operating system cannot compute the current slot, merely the current civic time.
+
+- The Plutus interface exposes the validity range as POSIX time, in the [`txInfoValidRange` field](https://plutus.cardano.intersectmbo.org/haddock/latest/plutus-ledger-api/PlutusLedgerApi-V1.html#t:TxInfo) (see this [blog post](https://iohk.io/en/blog/posts/2022/12/08/time-handling-on-cardano-part-2-use-cases/) for high-level background).
+  The Consensus Layer design would have been simpler if the Plutus API provided the validity range in terms of slots, but that ship has sailed.
+  (And developers writing Plutus scripts are almost certainly relieved that it did.)
+
+Because of those exceptions, the Consensus Layer must use the wall clock and/or the translation back and forth between slots and civic.
+Those uses are listed in the following table.
+
+ - Every use involves some translation, whether it be from slot-to-civic or vice versa (which is always the wall clock as a slot).
+
+ - This table excludes some uses that the Conensus Team is in the processing of removing (eg those that could be replaced by annotating validated headers with their slot onset).
+
+ - The rightmost column of the table judges whether the use obviously involves some entity that is obviously chain-dependent.
+   The column is for the benefit of a section below, but the key idea is that the translations are inherently chain-dependent, but users and most developers are blissfully unaware of that and, moreover, such a dependency is fundamentally contrary to human intuitions about time.
+
+ - The last few rows make explicit Consensus features that are not yet implemented but will also involve civic time.
+
+| Component | Use | Reads wall clock (whether translated to slot) | Which slots are translated to civic | Whether it's "obvious" that some involved entity is chain-dependent |
+| - | - | - | - | - |
+| ChainSync | enforce the clock skew bound | Yes (raw) | header's slot | Yes, headers |
+| ChainSel | define Plutus `txInfoValidRange`s when validating a block | No | validity range | Yes, blocks |
```

Comment:
Hmm, regarding ChainSel: For the lower bound, it is definitely true that the lower bound of the validity range is chain dependent, as it can be at most the current slot of the block that is currently being validated. However, for the upper bound, I don't think this is the case; we do not (yet) know what chain that slot will be on, ie we need a chain-independent (but stable) notion of time translation for upper bounds.

Reply:
I rephrased the column's semantics a bit. I think it's clearer now (the fact that chain-dependent translations are sufficient for things that are "chain-independent" precisely because they themselves identify which chain to use was a brain bender!)

Thanks for raising this. With the new column: the validating range is in some block, so that block's predecessor must agree to translate that validity range, and its translation is the only one that matters.

Reply:
Ah thanks, that makes a lot of sense, thanks for the rephrasing :+1: